### PR TITLE
chore(CPlusPlus): added search insert position

### PR DESCRIPTION
**Pull request by samughodake**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

**Review by Capt-Madhan — DISMISSED** [review]

**Review by Ashborn-SM — CHANGES_REQUESTED** [review]
@samughodake please do the changes mentioned.

**Comment by samughodake** [comment]
Hello, @Ashborn-SM 
I made all the desired changes.
Thanks.

File: algorithms/CPlusPlus/Arrays/search-insert-position.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    cin>>n;
+    cout<<"Enter the array elements\n";
+    vector<int> nums(n);
+    for(int i=0;i<n;i++){
+        cin>>nums[i];
+    }
+    cout<<"Enter the target element\n";
+    int target;
+    cin>>target;
+    cout<<"The target element can be inserted at: "<<searchInsert(nums,target);
```
Comment (Ashborn-SM) [inline algorithms/CPlusPlus/Arrays/search-insert-position.cpp:53]: The text here can be misleading as it prints `The target element can be inserted at: (some_index)` regardless of whether the target is present or not.

You don't want to print this if the target is present, in such cases a valid statement will be 
`The location of the target is (some_index)`.

My suggestion is to have a valid statement for both the cases(when the target is present & when the target is not present), you can make changes in the function itself accordingly.

**Review by Ashborn-SM — APPROVED** [review]
Good Work!!

**Pull request merged**